File: Restaurant-Backend/Entities/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Restaurant_Backend.Entities;

/// <summary>
/// Represents a product of the restaurant.
/// </summary>
[Table("Products")]
public class Product : EntityBase
{
    public string Name { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string? Description { get; set; }
    public bool IsAvailable { get; set; }
}

File: restaurant-client/src/types/product.ts
import type { EntityBase } from "./entityBase";

export interface Product extends EntityBase {
    name: string;
    price: number;
    description?: string;
    isAvailable: boolean;
  }

export interface ProductRequest {
    name: string;
    price: number;
    description?: string;
    isAvailable: boolean;
  }
  
  export interface ProductResponse {
    id: string;
    name: string;
    price: number;
    description?: string;
    isAvailable: boolean;
  }